crates/terminal_view/src/terminal_panel.rs

```diff
@@ -350,24 +350,37 @@ impl TerminalPanel {
         let mut spawn_task = spawn_in_terminal.clone();
         // Set up shell args unconditionally, as tasks are always spawned inside of a shell.
         let Some((shell, mut user_args)) = (match TerminalSettings::get_global(cx).shell.clone() {
-            Shell::System => std::env::var("SHELL").ok().map(|shell| (shell, Vec::new())),
+            Shell::System => Shell::retrieve_system_shell().map(|shell| (shell, Vec::new())),
             Shell::Program(shell) => Some((shell, Vec::new())),
             Shell::WithArguments { program, args } => Some((program, args)),
         }) else {
             return;
         };
 
-        spawn_task.command_label = format!("{shell} -i -c `{}`", spawn_task.command_label);
+        #[cfg(not(target_os = "windows"))]
+        {
+            spawn_task.command_label = format!("{shell} -i -c `{}`", spawn_task.command_label);
+        }
+        #[cfg(target_os = "windows")]
+        {
+            spawn_task.command_label = format!("{shell} -C `{}`", spawn_task.command_label);
+        }
         let task_command = std::mem::replace(&mut spawn_task.command, shell);
         let task_args = std::mem::take(&mut spawn_task.args);
         let combined_command = task_args
             .into_iter()
             .fold(task_command, |mut command, arg| {
                 command.push(' ');
+                #[cfg(not(target_os = "windows"))]
                 command.push_str(&arg);
+                #[cfg(target_os = "windows")]
+                command.push_str(&task::to_powershell_variable(arg));
```

Comment:
The args are parsed here:
https://github.com/zed-industries/zed/blob/7df8b6fe10deba1b74196f9c65a0f55a4ad7c185/crates/task/src/task_template.rs#L152
So I think it's safe to perform such operation here, so that `pwsh` can act normally.